### ISSUE TITLE
Displaying back button in the scaffold

### ISSUE THREAD

**Issue by 5AbhishekSaxena** [raamcosta/compose-destinations#46]
This is more of a feature request and less of a bug/issue.

**Problem**
How to know that the current screen is a top screen and there is no need to show back button. The counter-part of this issue is also true, which would be how to know the current screen has a predecessor or not.

**Requested feature**
There should be a way to know when to show the back button in the scaffold, this way we can show the back button based on the graph dynamically instead of hard-coding it in the scaffold.

**Suggestion**
Add a method in `Navigator` to check if the current screen has a predecessor or not, if it has then it can return true, else false.

**Comment by raamcosta** [comment]
Hi @5AbhishekSaxena ! 👋 

Do you know if you can do this with normal Jetpack Compose Navigation? If so, how?
That would help me 🙂

**Comment by 5AbhishekSaxena** [comment]
Hi @raamcosta 

No, I haven't tried it with Jetpack Compose Navigation, but I have done it with the Jetpack Navigation library, maybe that'll give you some insights.

Please refer to these links
- Update UI components with NavigationUI - https://developer.android.com/guide/navigation/navigation-ui
- NavigationUI - https://developer.android.com/reference/androidx/navigation/ui/NavigationUI

> The Navigation component includes a NavigationUI class. This class contains static methods that manage navigation with the top app bar, the navigation drawer, and the bottom navigation.

With the help of `NavigationUI` API, we can manage the content of the UI components w.r.t navigation from a central location like `MainActivity`.

**Comment by kkovach** [comment]
I'm currently trying to convert my app to use compose-destinations and I just got to the point where I want to do this as well. I'm not sure this is the best solution but this appears to be working for me right now... I stole the DestinationsSampleScaffold and add the following to it...

```
topBar = { destination ->
    TopAppBar(
        title = { Text(stringResource(id = destination.requireTitle)) },
        navigationIcon = {
            when(destination) {
                AddUserScreenDestination -> {
                    IconButton(onClick = {
                        navController.popBackStack()
                    }) {
                        Icon(Icons.Filled.ArrowBack, "")
                    }
                }
                else -> {
                    IconButton(onClick = {
                        coroutineScope.launch {
                            scaffoldState.drawerState.open()
                        }
                    }) {
                        Icon(Icons.Filled.Menu, "")
                    }
                }
            }
        }
    )
},
```

This raises a larger question that I had after looking through the sample app. What is the best way to modify the Scaffold on certain screens with respect to your sample app? What if I want the same general Scaffold with a different TopBar like I have above or a different BottomBar?

Currently, I only have the single Scaffold in the Main ComponentActivity and initial Composable and everything thus far has been within that. Thanks!

**Comment by raamcosta** [comment]
Hi @kkovach !

The general advice I've seen around (for Official Compose Navigation which should be the same for us too) is that people should have one Scaffold at the top level and all screens are shown inside it.
So I think your approach is going in the right direction: you check what destination you're currently at, and adapt the elements of the Scaffold as you wish to.
In fact, I do that also on the sample when the bottom bar updates depending on the current destination on the screen. Given, that is a simpler example, but the concept is the same.

As for any other support for this issue, I don't see myself implementing this any time soon. I feel like there are easy ways to do this and I don't even dislike them, so yeah.
You can define a list of Destinations on which you don't want to show the back button, and check if the current belongs to that, if so don't show it, if it doesn't then show the back.
You can also use multi nav back stack entries, like the [tivi fork](https://github.com/raamcosta/tivi) shows and when you are at the root of your nav graph, hide the back button, etc.
Basically, you need to figure out what is the best way you have to decide whether or not to show the back arrow, and just implement it. The tools are available 🙂 

Having said that, I will close this issue. The only way I see myself doing something is if Google implements something for their official library, in which case, I might adapt it to fit our APIs. Otherwise, I think I will not venture here 🙂

**Comment by raamcosta** [comment]
Let me know if this is helpful at all. I am eager to help you guys further if there is any other related point. 
Peace! ✌️
